Added Cybersuit Corp set (was Corpo World)

blackshadowshade opened the pull request:
I've updated this pull request to be mergeable once we are happy with it. Fixes #2664.

~This is the initial testing pull request addressing #2664. @cgolubi1, please deploy a test site for this when you can, so that testing can begin.~

~This is NOT TO BE MERGED, and I'm deliberately allowing the unit tests to fail until testing is completed.~

cgolubi1 commented:
Here you go - http://temporary.dev.buttonweavers.com/ui/buttons.html?set=Corpo%20World

cgolubi1 commented:
I'm setting up a replay site too, and i'll just have it play a bunch of novel Corpo World vs any games to make sure we don't have any skill combos in these buttons that cause crashes or stuck games.

cgolubi1 commented:
Replay testing played something like 3400 games, which should have been enough to catch obvious crash bugs over such a small number of buttons.  Calling that good.

TheOrgg commented:
The test site seems like it is messed up at the moment.  I also posted this on the Issue when I thought it was just me.  Anno has trouble with it, too.

cgolubi1 commented:
Looks like mysqld got OOM-killed:

```
[Sun Jan 31 06:53:29 2021] Out of memory: Kill process 21828 (mysqld) score 303 or sacrifice child
[Sun Jan 31 06:53:29 2021] Killed process 21828 (mysqld) total-vm:1121964kB, anon-rss:182280kB, file-rss:0kB
```

cgolubi1 commented:
I don't have an explanation for why that happened.  It should be back online now.

blackshadowshade commented:
I've added a new database update for the requested changes.

cgolubi1 commented:
Sorry to be so slow to update this; i missed the February commit.  Working on it now.

cgolubi1 commented:
Site updated; sorry to be slow.

blackshadowshade commented:
Sorry, left a message in #2664, but meant it to be here:

@cgolubi1, I've added an update file that adds these new testing recipes to the database as Fiona1, Nadia1, Gavin1, Fiona2, Nadia2, Gavin2.

I haven't changed data.button.sql, I'll clean all this up when testing is done and I recreate the files from scratch.

TheOrgg commented:
These are mostly finished, with one or two more characters to test, and I'm uploading them for putting into the test site to take a look at it and see what needs adjusting, if anything, to the graphical elements of the buttons.

![150GavinBlank](https://user-images.githubusercontent.com/44233993/124971034-58cd6b80-dfee-11eb-9898-bc1dc7cda005.png)
![150pxAndrew](https://user-images.githubusercontent.com/44233993/124971038-59660200-dfee-11eb-8b60-ae86d2b8dbe0.png)
![150pxChad](https://user-images.githubusercontent.com/44233993/124971041-59660200-dfee-11eb-9642-d968f09e5dfe.png)
![150pxFiona](https://user-images.githubusercontent.com/44233993/124971042-59fe9880-dfee-11eb-8ca0-92ac8c54d472.png)
![150pxIsabel](https://user-images.githubusercontent.com/44233993/124971043-59fe9880-dfee-11eb-922a-0c4848a2dd4c.png)
![150pxMonique](https://user-images.githubusercontent.com/44233993/124971044-59fe9880-dfee-11eb-82d2-0f4500a5b374.png)
![150pxNadia](https://user-images.githubusercontent.com/44233993/124971046-59fe9880-dfee-11eb-9c75-2df2165fd7c5.png)
![150pxSachin](https://user-images.githubusercontent.com/44233993/124971047-5a972f00-dfee-11eb-9c0e-7cedfc282002.png)

TheOrgg commented:
Final 150px images for the set when it goes live.

![150pxSachin](https://user-images.githubusercontent.com/44233993/130150220-030296bf-cf6b-430c-9943-73ff5f90e899.png)
![150pxAndrew](https://user-images.githubusercontent.com/44233993/130150222-cadb21b2-d503-4535-a003-c077ac8e0f43.png)
![150pxChad](https://user-images.githubusercontent.com/44233993/130150223-4bf58500-1e76-4d5c-86b0-3f499dae957e.png)
![150pxFiona](https://user-images.githubusercontent.com/44233993/130150224-ff005a61-2085-4c31-b287-d696df8241c1.png)
![150pxGavin](https://user-images.githubusercontent.com/44233993/130150225-6558a83b-46b9-42da-8889-be7a876afc34.png)
![150pxIsabel](https://user-images.githubusercontent.com/44233993/130150227-e4dba258-fb3d-428a-bf04-8cc881be69b2.png)
![150pxMonique](https://user-images.githubusercontent.com/44233993/130150228-87efcc48-1037-411d-9e58-79d9b1a3ab22.png)
![150pxNadia](https://user-images.githubusercontent.com/44233993/130150229-0729bb1a-e498-4527-bd43-a61df47c4845.png)

blackshadowshade commented:
Argh, no 1337!  XD

TheOrgg commented:
That's up to A_D, ain't it?  Haven't heard back since the last email, but he did confirm that he's been really busy.

blackshadowshade commented:
Any confirmation of the set name yet?

TheOrgg commented:
Not yet.  Just set up a game to get confirmation.

TheOrgg commented:
Anno wants non-leet CyberSuit Corp.

blackshadowshade commented:
This pull request should be ready for final review. It will conflict with any other pull request that changes the number of playable buttons or button sets.

TheOrgg commented:
So CyberSuit will be included with the giant set update!  Hooray!

blackshadowshade commented:
I've added the flavour text in an additional database update, and also updated data.button.sql.

No rebase yet, so that we can easily update the testing site.

blackshadowshade commented:
This has been rebased to resolve conflicts created by merging 2021 Fanatics.

cgolubi1 commented:
So, there was a dev site for this, but AWS stopped it awhile ago.  I could:
* reboot it, update the DNS, and apply the new DB update
* get rid of it and stand up a new dev site

The second of those is probably slightly easier, but neither is particularly hard.  Anyone have a preference?

TheOrgg commented:
Are they not ready to go to the Live site?

cgolubi1 commented on deploy/database/updates/02664_corpo_world_01.sql:
```diff
@@ -0,0 +1,12 @@
+INSERT INTO buttonset (id, name, sort_order) VALUES
+(86, 'Corpo World', 8600);
```
This is wrong --- it still has the old set name.

cgolubi1 commented:
DB update-vs-rebuild test failed --- see inline comment.  Fix that, and let me know if you (Shadowshade) are good with this being merged once that test passes.

blackshadowshade commented:
I've fixed the update file, that should now work.

I believe that we haven't run any replay testing on this set, so I think that's still waiting to happen.

cgolubi1 commented:
Sounds good, will do.  I don't have any specific skill interaction concerns with this set, but if i don't have to guess, i don't have to guess wrong.

cgolubi1 commented:
Verified that DB updates are fine now.

cgolubi1 commented:
Okay, this has played 7200 new games since yesterday, and hasn't crashed.  Good enough?  Should we merge it?  If not, remind me what else has to happen.

blackshadowshade commented:
I'm good to merge this, then I'll update the other pull requests to deal with the resulting merge conflicts.
